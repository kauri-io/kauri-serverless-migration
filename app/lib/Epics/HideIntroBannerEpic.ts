import cookie from 'cookie'
import { filter, mapTo, map } from 'rxjs/operators'
import { ActionsObservable } from 'redux-observable';

const HIDE_INTRO_BANNER: string = 'HIDE_INTRO_BANNER'
export const HIDE_INTRO_BANNER_SUCCESS: string = 'HIDE_INTRO_BANNER_SUCCESS'

export const hideIntroBannerAction = (): IHideIntroBannerAction => ({
    type: HIDE_INTRO_BANNER,
})

interface IHideIntroBannerAction {
    type: string
}

export default (action$: ActionsObservable<IHideIntroBannerAction>) =>
    action$.pipe(
        filter(action => action.type === HIDE_INTRO_BANNER),
        map(() => cookie.serialize('HIDE_INTRO_BANNER', 'true', {
            maxAge: 30 * 24 * 60 * 60 * 60, // 30 days
            })
        ),
        mapTo({ type: HIDE_INTRO_BANNER_SUCCESS })
    )