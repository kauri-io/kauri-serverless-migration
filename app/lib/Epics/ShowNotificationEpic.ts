import { ActionsObservable } from 'redux-observable'
import { filter, map, ignoreElements } from 'rxjs/operators'
import { notification } from 'antd'

export const SHOW_NOTIFICATION: string = 'SHOW_NOTIFICATION'

export const showNotificationAction = (
    payload: IShowNotificationPayload
): IShowNotificationAction => ({
    type: SHOW_NOTIFICATION,
    payload,
})

export interface IShowNotificationPayload {
    notificationType: 'success' | 'info' | 'warning' | 'error'
    message: string
    description: string
}

export interface IShowNotificationAction {
    type: string
    payload: IShowNotificationPayload
}

const openNotificationWithIcon = ({
    notificationType,
    message,
    description,
}: IShowNotificationPayload): void =>
    notification[notificationType]({
        placement: 'topLeft',
        message,
        description,
    })

export default (action$: ActionsObservable<IShowNotificationAction>) =>
    action$.pipe(
        filter(action => action.type === SHOW_NOTIFICATION),
        map(({ payload }) => openNotificationWithIcon(payload)),
        ignoreElements()
    )
