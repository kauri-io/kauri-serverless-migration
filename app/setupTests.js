import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
