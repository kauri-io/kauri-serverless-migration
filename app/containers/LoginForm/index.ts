import { compose } from 'recompose'
import { connect } from 'react-redux'
import {
    registerAction,
    IRegisterAction,
    IRegisterActionPayload,
} from './Module'
import { showNotificationAction } from '../../lib/Epics/ShowNotificationEpic'
import { withFormik } from 'formik'
import View from './View'

export type FormState = {
    email: string
}

const mapStateToProps = () => ({})

interface IProps {
    registerAction: (payload: IRegisterActionPayload) => IRegisterAction
    callback?: any
}

interface IHandleSumit {
    props: IProps
    resetForm: any
}

export default compose(
    connect(
        mapStateToProps,
        { registerAction, showNotificationAction }
    ),
    withFormik({
        mapPropsToValues: () => ({}),
        handleSubmit: async (
            values: FormState,
            { props, resetForm }: IHandleSumit
        ) => {
            console.log(props)
            console.log('Received values of form: ', values)
            return props.registerAction({ ...values }, resetForm)
        },
    })
)(View)
