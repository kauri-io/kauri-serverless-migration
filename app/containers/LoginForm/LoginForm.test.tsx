import LoginForm from './index'
import { mountWithRedux } from '../../setupTests'
import { MockedProvider } from 'react-apollo/test-utils'
import testEpic from '../../lib/test-epic'
import loginEpic, { registerAction } from './Module'
import { showNotificationAction } from '../../lib/Epics/ShowNotificationEpic'

describe('containers/LoginForm', () => {
    it('should match snapshot', () => {
        const props = {
            registerAction: jest.fn(),
        }
        const wrapper = mountWithRedux(
            <MockedProvider>
                <LoginForm {...props} />
            </MockedProvider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
