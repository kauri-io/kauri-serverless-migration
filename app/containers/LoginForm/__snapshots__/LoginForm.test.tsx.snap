// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/LoginForm should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {},
        "palette": Object {
          "action": Object {
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "hover": "rgba(0, 0, 0, 0.08)",
            "hoverOpacity": 0.08,
            "selected": "rgba(0, 0, 0, 0.14)",
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#f5f5f5",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#1E2428",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "primary": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "rgb(7, 118, 93)",
            "light": "rgb(59, 186, 158)",
            "main": "#0BA986",
          },
          "secondary": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "rgb(178, 178, 178)",
            "light": "rgb(255, 255, 255)",
            "main": "#ffffff",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12)",
          "0px 1px 5px 0px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 3px 1px -2px rgba(0,0,0,0.12)",
          "0px 1px 8px 0px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 3px 3px -2px rgba(0,0,0,0.12)",
          "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
          "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
          "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
          "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
          "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
          "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
          "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
          "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
          "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
          "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
          "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
          "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
          "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
          "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
          "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
        ],
        "shape": Object {
          "borderRadius": 4,
        },
        "spacing": [Function],
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": 1.5,
          },
          "body2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "letterSpacing": "0.01071em",
            "lineHeight": 1.43,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "letterSpacing": "0.02857em",
            "lineHeight": 1.75,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "letterSpacing": "0.03333em",
            "lineHeight": 1.66,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 14,
          "fontWeightBold": 700,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "h1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3.75rem",
            "fontWeight": 300,
            "letterSpacing": "-0.00833em",
            "lineHeight": 1,
          },
          "h3": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3rem",
            "fontWeight": 400,
            "letterSpacing": "0em",
            "lineHeight": 1.04,
          },
          "h4": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "letterSpacing": "0.00735em",
            "lineHeight": 1.17,
          },
          "h5": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "letterSpacing": "0em",
            "lineHeight": 1.33,
          },
          "h6": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.25rem",
            "fontWeight": 500,
            "letterSpacing": "0.0075em",
            "lineHeight": 1.6,
          },
          "htmlFontSize": 16,
          "overline": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "letterSpacing": "0.08333em",
            "lineHeight": 2.66,
            "textTransform": "uppercase",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subtitle1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": 1.75,
          },
          "subtitle2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "letterSpacing": "0.00714em",
            "lineHeight": 1.57,
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "tooltip": 1500,
        },
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "DEFAULT_CARD_WIDTH": 288,
          "MAIN_BODY_WIDTH": 1242,
          "bg": Object {
            "bgPrimary": "#1E2428",
            "bgSecondary": "#1E3D3B",
            "borderTextArea": "#979797",
            "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
            "disabledBackgroundColor": "#D6D6D6",
            "disabledTextColor": "#9B9B9B",
            "divider": "#EBEBEB",
            "errorRedColor": "#C03030",
            "hoverTextColor": "#108E72",
            "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
            "paddingTop": "2.5em",
            "primary": "#0BA986",
            "primaryColor": "#0BA986",
            "primaryDark": "#11856B",
            "primaryTextColor": "#1E2428",
            "secondary": "#11856B",
            "secondaryBlue": "rgb(81, 123, 162, 1)",
            "secondaryBlueDark": "rgb(81, 118, 142, 1)",
            "secondaryColor": "#1E3D3B",
            "secondaryTextColor": "#283035",
            "tertiaryBackgroundColor": "#F7F7F7",
            "textPrimary": "#1E2428",
            "white": "#FFFFFF",
          },
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "breakpoints": Array [
            "500px",
            "52em",
            "1242px",
          ],
          "colors": Object {
            "bgPrimary": "#1E2428",
            "bgSecondary": "#1E3D3B",
            "borderTextArea": "#979797",
            "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
            "disabledBackgroundColor": "#D6D6D6",
            "disabledTextColor": "#9B9B9B",
            "divider": "#EBEBEB",
            "errorRedColor": "#C03030",
            "hoverTextColor": "#108E72",
            "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
            "paddingTop": "2.5em",
            "primary": "#0BA986",
            "primaryColor": "#0BA986",
            "primaryDark": "#11856B",
            "primaryTextColor": "#1E2428",
            "secondary": "#11856B",
            "secondaryBlue": "rgb(81, 123, 162, 1)",
            "secondaryBlueDark": "rgb(81, 118, 142, 1)",
            "secondaryColor": "#1E3D3B",
            "secondaryTextColor": "#283035",
            "tertiaryBackgroundColor": "#F7F7F7",
            "textPrimary": "#1E2428",
            "white": "#FFFFFF",
          },
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "fontSizes": Array [
            11,
            13,
            14,
            16,
            18,
            20,
            22,
            28,
            32,
          ],
          "fontWeight": Array [
            300,
            "normal",
            500,
            700,
          ],
          "hoverTextColor": "#108E72",
          "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
          "padding": "calc((100vw - 1242px) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "space": Array [
            0,
            10,
            15,
            30,
            60,
          ],
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        }
      }
    >
      <MockedProvider
        addTypename={true}
      >
        <ApolloProvider
          client={
            ApolloClient {
              "__operations_cache__": Map {},
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "clearStoreCallbacks": Array [],
              "clientAwareness": Object {},
              "defaultOptions": Object {},
              "disableNetworkFetches": false,
              "link": ApolloLink {
                "request": [Function],
              },
              "mutate": [Function],
              "query": [Function],
              "queryDeduplication": true,
              "reFetchObservableQueries": [Function],
              "resetStore": [Function],
              "resetStoreCallbacks": Array [],
              "ssrMode": false,
              "store": DataStore {
                "cache": InMemoryCache {
                  "addTypename": true,
                  "cacheKeyRoot": KeyTrie {
                    "weakness": true,
                  },
                  "config": Object {
                    "addTypename": true,
                    "dataIdFromObject": [Function],
                    "fragmentMatcher": HeuristicFragmentMatcher {},
                    "freezeResults": false,
                    "resultCaching": true,
                  },
                  "data": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "maybeBroadcastWatch": [Function],
                  "optimisticData": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "silenceBroadcast": false,
                  "storeReader": StoreReader {
                    "executeSelectionSet": [Function],
                    "executeStoreQuery": [Function],
                    "executeSubSelectedArray": [Function],
                    "freezeResults": false,
                  },
                  "storeWriter": StoreWriter {},
                  "typenameDocumentCache": Map {},
                  "watches": Set {},
                },
              },
              "version": "2.4.13",
              "watchQuery": [Function],
            }
          }
        >
          <Connect(WithFormik(LoginFormContainer))
            registerAction={[MockFunction]}
          >
            <WithFormik(LoginFormContainer)
              registerAction={[Function]}
              showNotificationAction={[Function]}
            >
              <Formik
                enableReinitialize={false}
                handleSubmit={[Function]}
                initialValues={Object {}}
                isInitialValid={false}
                onSubmit={[Function]}
                registerAction={[Function]}
                render={[Function]}
                showNotificationAction={[Function]}
                validateOnBlur={true}
                validateOnChange={true}
              >
                <LoginFormContainer
                  dirty={false}
                  errors={Object {}}
                  handleBlur={[Function]}
                  handleChange={[Function]}
                  handleReset={[Function]}
                  handleSubmit={[Function]}
                  initialValues={Object {}}
                  isSubmitting={false}
                  isValid={false}
                  isValidating={false}
                  registerAction={[Function]}
                  registerField={[Function]}
                  resetForm={[Function]}
                  setError={[Function]}
                  setErrors={[Function]}
                  setFieldError={[Function]}
                  setFieldTouched={[Function]}
                  setFieldValue={[Function]}
                  setFormikState={[Function]}
                  setStatus={[Function]}
                  setSubmitting={[Function]}
                  setTouched={[Function]}
                  setValues={[Function]}
                  showNotificationAction={[Function]}
                  submitCount={0}
                  submitForm={[Function]}
                  touched={Object {}}
                  unregisterField={[Function]}
                  validateField={[Function]}
                  validateForm={[Function]}
                  validateOnBlur={true}
                  validateOnChange={true}
                  values={Object {}}
                >
                  <Web3Unavailable>
                    <View__LoginContainer>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "View__LoginContainer-v8ha34-1",
                              "isStatic": false,
                              "lastClassName": "ikCtyG",
                              "rules": Array [
                                "display:flex;flex-direction:column;align-items:center;text-align:center;justify-content:center;& > *{margin:",
                                [Function],
                                "px;}& #mc_embed_signup{width:400px;color:",
                                [Function],
                                ";padding:",
                                [Function],
                                "px;border-radius:4px;margin-top:",
                                [Function],
                                "px;}",
                              ],
                            },
                            "displayName": "View__LoginContainer",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "View__LoginContainer-v8ha34-1",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="View__LoginContainer-v8ha34-1 ikCtyG"
                        >
                          <Typography__styledComponent
                            color="white"
                          >
                            <StyledComponent
                              color="white"
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Typography__styledComponent-sc-36xkf6-5",
                                    "isStatic": false,
                                    "lastClassName": "jNVtUZ",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";margin:0px;font-weight:",
                                      "500",
                                      ";",
                                      ";font-size:",
                                      "20",
                                      "px;text-transform:",
                                      [Function],
                                      ";text-align:",
                                      [Function],
                                      ";:hover{color:",
                                      [Function],
                                      ";cursor:",
                                      [Function],
                                      ";}",
                                    ],
                                  },
                                  "displayName": "Typography__styledComponent",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="Typography__styledComponent-sc-36xkf6-5 jNVtUZ"
                                color="white"
                              >
                                Web3 Sign in
                              </span>
                            </StyledComponent>
                          </Typography__styledComponent>
                          <Typography__component
                            color="white"
                          >
                            <StyledComponent
                              color="white"
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Typography__component-sc-36xkf6-3",
                                    "isStatic": true,
                                    "lastClassName": "cGrhPv",
                                    "rules": Array [
                                      "font-size:17px;font-weight:normal;letter-spacing:-0.1px;line-height:24px;",
                                    ],
                                  },
                                  "displayName": "Typography__component",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Typography__component-sc-36xkf6-3",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="Typography__component-sc-36xkf6-3 cGrhPv"
                                color="white"
                              >
                                You need the MetaMask extension to use Kauri. (MetaMask supports Chrome, Firefox, Opera)
                              </span>
                            </StyledComponent>
                          </Typography__component>
                          <View__Image
                            src="/static/images/metamask/avatar.png"
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "View__Image-v8ha34-2",
                                    "isStatic": false,
                                    "lastClassName": "jKbdBy",
                                    "rules": Array [
                                      "width:100px;height:91px;margin:",
                                      [Function],
                                      "px;",
                                    ],
                                  },
                                  "displayName": "View__Image",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "View__Image-v8ha34-2",
                                  "target": "img",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              src="/static/images/metamask/avatar.png"
                            >
                              <img
                                className="View__Image-v8ha34-2 jKbdBy"
                                src="/static/images/metamask/avatar.png"
                              />
                            </StyledComponent>
                          </View__Image>
                          <a
                            href="https://metamask.io"
                            target="_blank"
                          >
                            https://metamask.io
                          </a>
                          <div>
                            Here for the content? Sign up for a newsletter below and receive the latest Web3 tutorials, project announcements, and articles every 2 weeks!
                          </div>
                          <link
                            href="//cdn-images.mailchimp.com/embedcode/classic-10_7.css"
                            rel="stylesheet"
                            type="text/css"
                          />
                          <style
                            dangerouslySetInnerHTML={
                              Object {
                                "__html": "	#mc_embed_signup{background:#fff; clear:left; font:14px Helvetica,Arial,sans-serif; }",
                              }
                            }
                            type="text/css"
                          />
                          <div
                            id="mc_embed_signup"
                          >
                            <form
                              action="https://kauri.us17.list-manage.com/subscribe/post?u=e46233ccfd6bb938ab7cbb5a3&id=f49f81a2a9"
                              className="validate"
                              id="mc-embedded-subscribe-form"
                              method="post"
                              name="mc-embedded-subscribe-form"
                              noValidate={true}
                              target="_blank"
                            >
                              <div
                                id="mc_embed_signup_scroll"
                              >
                                <div
                                  className="indicates-required"
                                >
                                  <span
                                    className="asterisk"
                                  >
                                    *
                                  </span>
                                   indicates required
                                </div>
                                <div
                                  className="mc-field-group"
                                >
                                  <label
                                    htmlFor="mce-EMAIL"
                                  >
                                    Email Address 
                                    <span
                                      className="asterisk"
                                    >
                                      *
                                    </span>
                                  </label>
                                  <input
                                    className="required email"
                                    id="mce-EMAIL"
                                    name="EMAIL"
                                    type="email"
                                  />
                                </div>
                                <div
                                  className="clear"
                                  id="mce-responses"
                                >
                                  <div
                                    className="response"
                                    id="mce-error-response"
                                    style={
                                      Object {
                                        "display": "none",
                                      }
                                    }
                                  />
                                  <div
                                    className="response"
                                    id="mce-success-response"
                                    style={
                                      Object {
                                        "display": "none",
                                      }
                                    }
                                  />
                                </div>
                                <div
                                  aria-hidden="true"
                                  style={
                                    Object {
                                      "left": "-5000px",
                                      "position": "absolute",
                                    }
                                  }
                                >
                                  <input
                                    defaultValue=""
                                    name="b_e46233ccfd6bb938ab7cbb5a3_f49f81a2a9"
                                    tabIndex={-1}
                                    type="text"
                                  />
                                </div>
                                <div
                                  className="clear"
                                >
                                  <input
                                    className="button"
                                    defaultValue="Subscribe"
                                    id="mc-embedded-subscribe"
                                    name="subscribe"
                                    type="submit"
                                  />
                                </div>
                              </div>
                            </form>
                          </div>
                          <script
                            src="//s3.amazonaws.com/downloads.mailchimp.com/js/mc-validate.js"
                            type="text/javascript"
                          />
                          <script
                            dangerouslySetInnerHTML={
                              Object {
                                "__html": "(function($) {window.fnames = new Array(); window.ftypes = new Array();fnames[0]='EMAIL';ftypes[0]='email';}(jQuery));var $mcj = jQuery.noConflict(true);",
                              }
                            }
                            type="text/javascript"
                          />
                        </div>
                      </StyledComponent>
                    </View__LoginContainer>
                  </Web3Unavailable>
                </LoginFormContainer>
              </Formik>
            </WithFormik(LoginFormContainer)>
          </Connect(WithFormik(LoginFormContainer))>
        </ApolloProvider>
      </MockedProvider>
    </ThemeProvider>
  </ThemeProvider>
</Provider>
`;
